tests/Agency/StoreTestAPI.cpp

```diff
@@ -0,0 +1,973 @@
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Ignacio Rodriguez
+/// @author Max Neunhoeffer
+/// @author Copyright 2021, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include <algorithm>
+#include "gtest/gtest.h"
+
+#include "Agency/Store.h"
+#include "Mocks/Servers.h"
+
+#include <velocypack/Builder.h>
+#include <velocypack/Compare.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+namespace arangodb {
+namespace tests {
+namespace store_test_api {
+
+class StoreTestAPI : public ::testing::Test {
+ public:
+  StoreTestAPI() : _server(), _store(_server.server(), nullptr) {
+  }
+ protected:  
+  arangodb::tests::mocks::MockCoordinator _server;
+  arangodb::consensus::Store _store;
+
+  auto readAndCheck(std::string const &json) 
+  {
+    try {
+      auto q {VPackParser::fromJson(json)};
+      auto result {std::make_shared<VPackBuilder>()};
+      _store.read(q, result);
+      return result;
+    }
+    catch(std::exception &ex) {
+      throw std::runtime_error(std::string(ex.what()) + " while trying to read " + json);
+    }
+  }
+
+  auto write(std::string const &json)
+  {
+    try {
+      auto q {VPackParser::fromJson(json)};
+      return _store.applyTransactions(q);
+    }
+    catch(std::exception &err) {
+      throw std::runtime_error(std::string(err.what()) + " while parsing " + json );
+    }
+  }
+
+  template<typename TOstream, typename TValueContainer>
+  static TOstream& insert_value_array(TOstream &out, TValueContainer const &src) {
+    out << "[";
+    if (!src.empty()) {
+      std::copy(src.begin(), src.end()-1, std::ostream_iterator<std::string>(out, ", "));
+      out << src.back();
+    }
+    out << "]";
+    return out;
+  }
+
+  auto write(std::vector<std::vector<std::string>> const &operations) {
+    std::stringstream ss;
+    ss << "[";
+    if (!operations.empty()) {
+      insert_value_array(ss, operations.front());
+      for (auto oi {operations.begin() + 1}; oi != operations.end(); ++oi) {
+        insert_value_array(ss, *oi);
+      }
+    }
+    ss << "]";
+    return write(ss.str());
+  }
+
+  auto transactAndCheck(std::string const &json) {
```

Comment:
Name is misleading.

Reply:
No checks are done.

Reply:
Agreed. The semantics were changed to match the name in this case.